cmd/derper/bootstrap_dns.go

```diff
@@ -35,10 +46,32 @@ func refreshBootstrapDNS() {
 	if *bootstrapDNS == "" {
 		return
 	}
-	dnsEntries := make(map[string][]net.IP)
-	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	ctx, cancel := context.WithTimeout(context.Background(), refreshTimeout)
+	defer cancel()
+	dnsEntries := resolveList(ctx, strings.Split(*bootstrapDNS, ","))
+	j, err := json.MarshalIndent(dnsEntries, "", "\t")
+	if err != nil {
+		// leave the old values in place
+		return
+	}
+	dnsCache.Store(j)
+}
+
+func refreshUnpublishedDNS() {
+	if *unpublishedDNS == "" {
+		return
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), refreshTimeout)
 	defer cancel()
-	names := strings.Split(*bootstrapDNS, ",")
+
+	dnsEntries := resolveList(ctx, strings.Split(*unpublishedDNS, ","))
```

Comment:
keep in mind that strings.Split on an empty string will yield a 1-length slice of an empty string: https://go.dev/play/p/m5X9rfocwem

Reply:
Should be handled by the `if` statement at the top that exits if the string is empty

Reply:
whoops, missed that